### PR TITLE
feat(elements-angular): add icons to angular.json on `ng add`

### LINKED ISSUE #390: Angular Cli Integration
### Is your feature request related to a problem? Please describe.

In order to simply the setup, we should add Schematics for the `ng add` command. This helps us to automatically add the assets to the `angular.json` of the client application.

### Describe the solution you'd like

Add schematics for ng add and automate the icon assets tasks (https://angular.io/guide/schematics-for-libraries#schematics-for-libraries)

### Additional context

Untested idea derived from https://github.com/ionic-team/ionic-framework/tree/master/angular/src/schematics:

`packages/elements-angular/elements/schematics/index.ts`
```typescript 
import { Rule, SchematicContext, SchematicsException, Tree } from '@angular-devkit/schematics';
import { chain } from '@angular-devkit/schematics';
import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks'

const CONFIG_PATH = 'angular.json';

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

function getAngularAppConfig(config: any, projectName: string): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  const projectConfig = config.projects[projectName];
  return projectConfig;
}

function addAsset(host: Tree, projectName: string, architect: string, asset: string | {glob: string; input: string; output: string}) {
  const config = readConfig(host);
  const appConfig = getAngularAppConfig(config, projectName);
  appConfig.architect[architect].options.assets.push(asset);
  writeConfig(host, config);
}

function installNodeDeps() {
  return (_host: Tree, context: SchematicContext) => {
    context.addTask(new NodePackageInstallTask());
  };
}

function addIcons(projectName: string): Rule {
  return (host: Tree) => {
    const inoIconsGlob = {
      glob: '**/*.svg',
      input: 'node_modules/@inovex.de/elements/dist/inovex-elements/ino-icon',
      output: 'ino-icon'
    };
    addAsset(host, projectName, 'build', inoIconsGlob);
    addAsset(host, projectName, 'test', inoIconsGlob);
    return host;
  };
}

// Just return the tree
export function ngAdd(options: any): Rule {
  return chain([
      addIcons(options.project),
      // install freshly added dependencie
      installNodeDeps()
  ]);
}
```

`packages/elements-angular/elements/schematics/collection.json`
```json
{
  "$schema": "../../node_modules/@angular-devkit/schematics/collection-schema.json",
  "schematics": {
    "ng-add": {
      "description": "Add inovex elements to your project",
      "factory": "./ng-add/index#ngAdd",
      "save": "dependencies"
    }
  }
```
}

### PR DESCRIPTION
**Pull request by janivo**
Closes #390 

## Proposed Changes

- Add schematic for angular CLI to automatically add our icons on `ng add @inovex.de/elements-angular`

### CODE REVIEW

File: packages/elements-angular/elements/schematics/collection.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "../node_modules/@angular-devkit/schematics/collection-schema.json",
+  "schematics": {
+    "ng-add": {
+      "description": "Add the inovex-elements to the project.",
+      "factory": "./ng-add/index#ngAdd",
```
Comment (pfecht) [inline packages/elements-angular/elements/schematics/collection.json:6]: ` "factory": "./ng-add/index#ngAdd"` (no trailing comma)

**Review by pfecht — CHANGES_REQUESTED** [review]
Nice 🔥  But I'm still not sure, how to test if the schematics actually works. I see the schematics folder in dist but how to apply it to a project? 

Could you add documentation on
* how to test the schematics during development? 
* update the framework integration of angular to `ng add`

**Comment by janivo** [comment]
> Nice 🔥 But I'm still not sure, how to test if the schematics actually works. I see the schematics folder in dist but how to apply it to a project?
> 
> Could you add documentation on
> 
> * how to test the schematics during development?
> * update the framework integration of angular to `ng add`

Yea, didn't even think about that. My bad here 🤦🏼‍♂️ 

I added some documentation, have a look at it again!

**Review by pfecht — APPROVED** [review]
Thanks! Clean docs now 🔥

File: packages/elements-angular/elements/README.md [hunk 2]
```diff
@@ -12,25 +12,35 @@ Please refer to the top level [README](../../../README.md) to see the available
...
+For a detailed explanation of schematics in Angular, see [this guide](https://angular.io/guide/schematics).
+
+At this moment, `ng-add` is the only schematic this library provides.
+All the related files are located under `/schematics`.
+
+In order to test the functionality of the schematics, we make use of our linking tool `yalc`:
+1. Add a new schematic or make changes to the existing one
+2. Run `publish:yalc` at the root of this project and select angular
+3. In your consuming project, install [yalc](https://github.com/wclr/yalc) and run `yalc link @inovex.de/elements-angular`
+4. Run `ng add @inovex.de/elements-angular`
```
Comment (Sl1nd) [inline packages/elements-angular/elements/README.md:38]: Does not work as described, even on starting from clean state.

grafik

Comment (janivo) [reply packages/elements-angular/elements/README.md:38]: I think I found the error cause: I did not check if the object exists in your `angular.json` before adding the icons.

Comment (Sl1nd) [reply packages/elements-angular/elements/README.md:38]: Tested. The error is fixed an the angular.json is edited appropriate.

**Review by Sl1nd — APPROVED** [review]

**Pull request merged**